src/DocBlock/TypeExpression.php

```diff
@@ -53,13 +53,13 @@ final class TypeExpression
                     (?<array_shape_start>(?i)(?:array|list|object)(?-i)\h*\{\h*)
                     (?<array_shape_inners>
                         (?<array_shape_inner>
-                            (?<array_shape_inner_key>(?:(?&constant)|(?&identifier)|(?&name))\h*\??\h*:\h*|)
+                            (?<array_shape_inner_key>(?:[^$\s,\{\|]+)\h*\??\h*:\h*|)
```

Comment:
the `(?&constant)|(?&identifier)` is/was here correctly

and maybe the fix with `|(?&name)` is not that bad

here is phpstan parser code for "array shape key": https://github.com/phpstan/phpdoc-parser/blob/1.25.0/src/Parser/TypeParser.php#L839 - it allows basically integer, quoted strings, and identifier

we have a little different definion of identifier, thus the difference

here is an example of wrong phpdoc: https://phpstan.org/r/5effc31f-ddff-4e77-ad01-d7844d09beec - we should not parse more, so I prefer to revert this line